pinot-broker/src/main/java/org/apache/pinot/broker/broker/AuthenticationFilter.java

```diff
@@ -82,9 +83,12 @@ public void filter(ContainerRequestContext requestContext)
 
     HttpRequesterIdentity httpRequestIdentity = HttpRequesterIdentity.fromRequest(request);
 
-    // default authorization handling
-    if (!accessControl.hasAccess(httpRequestIdentity)) {
-      throw new WebApplicationException("Failed access check for " + httpRequestIdentity.getEndpointUrl(),
+    AuthorizationResult authorizationResult = accessControl.authorize(httpRequestIdentity);
+
+    if (!authorizationResult.hasAccess()) {
+      throw new WebApplicationException(
+          "Failed access check for " + httpRequestIdentity.getEndpointUrl() + " ,with reason: "
```

Comment:
```suggestion
          "Failed access check for " + httpRequestIdentity.getEndpointUrl() + " with reason: "
```

Reply:
Also suggest checking if failure message is empty first, then decide whether to append the reason part. Same for other places